site_scons/site_tools/unittest.py

```diff
@@ -61,9 +63,30 @@ def __call__(self, target, source, env):
         out = str(target[0])
         exe = str(source[0])
 
-        try:
+        # On Mac OS X El Capitan we will need to side load the library load path
+        # and fix up the binary rpath
+        libpathstr = ""
+        if platform_default() == 'darwin':
+            # El Capitan is Darwin 15.x. We could simply always set this
+            # variable on OS X rather than restricting to version as
+            # in the future it will be more likely than not that we
+            # are running a SIP platform.
+            release_str = platform.release()
+            release_major = int(release_str.split('.')[0])
+            if release_major >= 15:
+                if "LSST_LIBRARY_PATH" in os.environ:
+                    os.environ["DYLD_LIBRARY_PATH"] = os.environ["LSST_LIBRARY_PATH"]
+
+            # We need to tell the tests where the uninstalled Qserv libraries
+            # are located because rpaths aren't (yet?) handled properly
+            # when the libraries and binaries are built on OS X.
+            os.environ["DYLD_LIBRARY_PATH"] = "build/:" + os.environ["DYLD_LIBRARY_PATH"]
```

Comment:
Also this code will be executed multiple times (once per unit test) so it will extend DYLD_LIBRARY_PATH in the environment every time in case the above `if` does not reset it.